pkg/kubectl/cmd/exec.go

```diff
@@ -159,7 +160,8 @@ func (p *ExecOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, argsIn []s
 		}
 		p.Command = argsIn
 	} else {
-		p.PodName = argsIn[0]
+		// Extract "pod/" from pod name.
+		p.PodName = strings.TrimPrefix(argsIn[0], "pod/")
```

Comment:
this should look more like what we do for `kubectl attach`, using resource builder (to support `foo`, `pod/foo`, `pods/foo`, `pods.v1/foo`, etc)

https://github.com/kubernetes/kubernetes/blob/97169b1512db0699bd4f3932ca9660e7659ba063/pkg/kubectl/cmd/attach/attach.go#L222-L245

(that will also mean moving the pod name lookup to the Run method)